I get the following compiler warning when I run the Stratum build:
```text
dfoster@dgfoster-mobl1:~/latest/stratum/stratum$ bazel build //stratum/hal/bin/tdi/es2k:stratum_es2k --define target=es2k
Starting local Bazel server and connecting to it...
DEBUG: /home/dfoster/latest/stratum/stratum/stratum/hal/lib/tdi/es2k/es2k.bzl:18:10: Target is: 'ES2K'
INFO: Analyzed target //stratum/hal/bin/tdi/es2k:stratum_es2k (225 packages loaded, 12692 targets configured).
INFO: Found 1 target...
INFO: From Compiling stratum/hal/lib/tdi/tdi_fixed_function_manager.cc:
stratum/hal/lib/tdi/tdi_fixed_function_manager.cc: In destructor 'stratum::hal::tdi::SadbConfigWrapper::~SadbConfigWrapper()':
stratum/hal/lib/tdi/tdi_fixed_function_manager.cc:25:41: warning: 'void* memset(void*, int, size_t)' clearing an object of type 'std::string' {aka 'class std::__cxx11::basic_string<char>'} with no trivial copy-assignment; use assignment or value-initialization instead [-Wclass-memaccess]
   25 |       memset(&key[0], 0, key->capacity());
      |                                         ^
In file included from /usr/include/c++/9/string:55,
                 from external/com_google_absl/absl/synchronization/mutex.h:63,
                 from ./stratum/hal/lib/tdi/tdi_fixed_function_manager.h:7,
                 from stratum/hal/lib/tdi/tdi_fixed_function_manager.cc:4:
/usr/include/c++/9/bits/basic_string.h:81:11: note: 'std::string' {aka 'class std::__cxx11::basic_string<char>'} declared here
   81 |     class basic_string
      |           ^~~~~~~~~~~~
Target //stratum/hal/bin/tdi/es2k:stratum_es2k up-to-date:
  bazel-bin/stratum/hal/bin/tdi/es2k/stratum_es2k
INFO: Elapsed time: 17.557s, Critical Path: 6.73s
INFO: 3 processes: 1 internal, 2 linux-sandbox.
INFO: Build completed successfully, 3 total actions
```
Any suggestions?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,10 @@ find_package(absl REQUIRED)
 find_package(gflags REQUIRED)
 find_package(glog REQUIRED)
 
+if(ES2K_TARGET)
+    find_package(OpenSSL REQUIRED)
+endif()
+
 set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
 
 # Suppress "warning: attribute ignored" on ABSL_MUST_USE_RESULT [[nodiscard]]
@@ -106,6 +110,13 @@ function(add_stratum_google_libs _TGT)
     )
 endfunction()
 
+# OpenSSL libraries
+function(add_stratum_openssl_libs _TGT)
+    if(ES2K_TARGET)
+        target_link_libraries(${_TGT} PUBLIC OpenSSL::Crypto)
+    endif()
+endfunction()
+
 # Protobuf libraries
 function(add_stratum_proto_libs _TGT)
     target_link_libraries(${_TGT} PUBLIC
@@ -120,6 +131,7 @@ endfunction()
 function(add_stratum_libraries _TGT)
     add_stratum_abseil_libs(${_TGT})
     add_stratum_google_libs(${_TGT})
+    add_stratum_openssl_libs(${_TGT})
     add_stratum_proto_libs(${_TGT})
     target_link_libraries(${_TGT} PUBLIC pthread)
 endfunction()

--- a/stratum/hal/lib/tdi/tdi_fixed_function_manager.cc
+++ b/stratum/hal/lib/tdi/tdi_fixed_function_manager.cc
@@ -11,6 +11,10 @@
 #include "stratum/hal/lib/tdi/utils.h"
 #include "stratum/lib/utils.h"
 
+extern "C" {
+#include "openssl/crypto.h"
+}
+
 namespace stratum {
 namespace hal {
 namespace tdi {
@@ -22,10 +26,7 @@ class SadbConfigWrapper {
   ~SadbConfigWrapper() {
     std::string* key = get_key_ptr();
     if (key) {
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wclass-memaccess"
-      memset(&key[0], 0, key->capacity());
-#pragma GCC diagnostic pop
+      OPENSSL_cleanse(&key[0], key->size());
     }
   }
 

--- a/stratum/hal/lib/tdi/tdi_fixed_function_manager.cc
+++ b/stratum/hal/lib/tdi/tdi_fixed_function_manager.cc
@@ -22,7 +22,10 @@ class SadbConfigWrapper {
   ~SadbConfigWrapper() {
     std::string* key = get_key_ptr();
     if (key) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
       memset(&key[0], 0, key->capacity());
+#pragma GCC diagnostic pop
     }
   }
 
@@ -129,8 +132,7 @@ TdiFixedFunctionManager::CreateInstance(OperationMode mode,
 ::util::Status TdiFixedFunctionManager::BuildSadbTableData(
     TdiSdeInterface::TableDataInterface* table_data,
     IPsecSADBConfig& sadb_config) {
-  // Wrapper object ensures that sadb_config is scrubbed on exit.
-  SadbConfigWrapper w(sadb_config);
+  SadbConfigWrapper sanitize_on_exit(sadb_config);
 
   CHECK_RETURN_IF_FALSE(table_data);
 

